C# references invalid string encoding.

<!-- Please provide a general summary of the issue, including what you're trying to accomplish. -->
This was found compiling our program codegen tests. 
## Steps to reproduce
<!-- Provide a link to a live example or an unambiguous set of steps to reproduce this bug. Tell us what you expected to happen, and what happened instead. Please include code to reproduce, if relevant. -->

1. `cd pkg/codegen/internal/test/testdata/functions-pp`
2. Fix the array so it's valid: this can be changing lines 9-13 to
```c#
        var joined = string.Join("-", new String[2]
        {
            "haha",
            "business",
        });
```
3. `dotnet build --nologo`


Expected: A clean compile
Actual: 
```
  Determining projects to restore...
  All projects are up-to-date for restore.
  Pulumi -> /Users/ianwahbe/Projects/pulumi/pulumi/sdk/dotnet/Pulumi/bin/Debug/netcoreapp3.1/Pulumi.dll
/Users/ianwahbe/Projects/pulumi/pulumi/pkg/codegen/internal/test/testdata/functions-pp/functions.cs(8,46): error CS0234: The type or namespace name 'UTF8' does not exist in the namespace 'System.Text' (are you missing an assembly reference?) [/Users/ianwahbe/Projects/pulumi/pulumi/pkg/codegen/internal/test/testdata/functions-pp/functions-pp.csproj]

Build FAILED.

/Users/ianwahbe/Projects/pulumi/pulumi/pkg/codegen/internal/test/testdata/functions-pp/functions.cs(8,46): error CS0234: The type or namespace name 'UTF8' does not exist in the namespace 'System.Text' (are you missing an assembly reference?) [/Users/ianwahbe/Projects/pulumi/pulumi/pkg/codegen/internal/test/testdata/functions-pp/functions-pp.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:02.86
```

--- a/pkg/codegen/dotnet/gen_program_expressions.go
+++ b/pkg/codegen/dotnet/gen_program_expressions.go
@@ -438,7 +438,7 @@ func (g *generator) GenFunctionCallExpression(w io.Writer, expr *model.FunctionC
 	case "split":
 		g.Fgenf(w, "%.20v.Split(%v)", expr.Args[1], expr.Args[0])
 	case "toBase64":
-		g.Fgenf(w, "Convert.ToBase64String(System.Text.UTF8.GetBytes(%v))", expr.Args[0])
+		g.Fgenf(w, "Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes(%v))", expr.Args[0])
 	case "toJSON":
 		g.Fgen(w, "JsonSerializer.Serialize(")
 		g.genDictionaryOrTuple(w, expr.Args[0])

--- a/pkg/codegen/testing/test/testdata/functions-pp/dotnet/functions.cs
+++ b/pkg/codegen/testing/test/testdata/functions-pp/dotnet/functions.cs
@@ -5,7 +5,7 @@ class MyStack : Stack
 {
     public MyStack()
     {
-        var encoded = Convert.ToBase64String(System.Text.UTF8.GetBytes("haha business"));
+        var encoded = Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes("haha business"));
         var joined = string.Join("-", 
         {
             "haha",

